repository.go

```diff
@@ -3,17 +3,21 @@ package goddd
 import (
 	"bytes"
 	"encoding/gob"
+	"errors"
 	"fmt"
 	"time"
 
 	"github.com/google/uuid"
 )
 
-type Repository interface {
-	Save(object DomainObject) error
-	Load(objectID string, object DomainObject) error
+var ConcurrencyError = errors.New("Concurrency error while saving")
```

Comment:
```suggestion
var ConcurrencyError = errors.New("concurrency error while saving")
```

Par convention le texte des erreurs doit être lowercase
https://google.github.io/styleguide/go/decisions.html#:~:text=Error%20strings%20should%20not%20be,%22Something%20bad%20happened.%22)